safehttp/mux.go

```diff
@@ -80,7 +80,8 @@ type ServeMux struct {
 	handlers map[string]methodHandler
 
 	// Maps interceptor key to interceptor.
-	interceps map[string]Interceptor
+	interceps      map[string]Interceptor
+	intercepsOrder []string
```

Comment:
Note: this is an artifact of having keyed interceptors. We've decided today that this should not be the case, so in a subsequent PR this should be changed to `interceptors []Interceptor`, or even better `interceptors []appliedInterceptor` where:

```
type appliedInterceptor struct {
  interceptor Interceptor
  cfg interface{}
}
```